packages/hint-compat-api/src/utils/parse-supports.ts

```diff
@@ -0,0 +1,196 @@
+export type Declaration = {
+    prop: string;
+    value: string;
+};
+
+export type DeclarationGroup = {
+    type: 'and' | 'or' | 'not';
+    nodes: (Declaration | DeclarationGroup)[];
+};
+
+type Token = '(' | ')' | 'not' | 'and' | 'or' | Declaration;
+
+class InvalidInputError extends Error {
```

Comment:
You have the same `InvalidInputError` in the file ` packages/hint-compat-api/src/utils/filter-supports.ts`, maybe you can move this to its own file and share it.

Reply:
By keeping them distinct classes, if `parse-supports.ts` accidentally lets one escape, it won't get trapped by `filter-supports.ts` so we can see the issue and fix it.

I can give them unique names if we want, but I don't think we should share a single definition as each component should own it's own "ignored" error type.

Reply:
Alternatively I could just add comment clarifying why the error type is not shared if that works for you.

Reply:
hehehe I saw the same name/code and I thought it was the same.
Maybe a comment for future refactors would be nice :)